### PR TITLE
Add support for Python 3.6.

### PR DESCRIPTION
**Pull request by tseaver**
I don't think Travis or Appveyor support 3.6 yet.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -14,7 +14,6 @@
 
 import os
 
-from setuptools import find_packages
```
Comment (dhermes) [inline setup.py]: Nice find. I noticed this a few weeks ago but never sent a PR. Shame

**Comment by tseaver** [comment]
@dhermes I'm contemplating trying to speed the build up by dropping our hacked-around namespace stuff:  a quick test seems to indicate that `pip install -e` now works.

**Comment by dhermes** [comment]
@tseaver I welcome the change. That'll just require lower bounds on `pip` / `setuptools` in our CircleCI / Travis / AppVeyor configs?

**Comment by lukesneeringer** [comment]
Travis and CircleCI both have Python 3.6 support now. :-)

**Comment by jonparrott** [comment]
Ya'll will probably need to wait on https://github.com/grpc/grpc/issues/9445 just like google-auth https://github.com/GoogleCloudPlatform/google-auth-library-python/pull/102

**Comment by lukesneeringer** [comment]
This is available now, but I am actually closing this because I am working on a more dramatic overhaul of our testing tooling.

**Pull request closed without merging**